crates/task-impls/src/quorum_proposal.rs

```diff
@@ -461,83 +448,111 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
                 }
                 Either::Left(_) => {
                     qc_dependency.mark_as_completed(event);
-                    info!(
-                        "Node {} Dependency QC is complete for view {}!",
-                        self.id, *view_number
-                    );
                 }
             },
             HotShotEvent::ViewSyncFinalizeCertificate2Recv(_) => {
                 view_sync_dependency.mark_as_completed(event);
-                info!(
-                    "Node {} Dependency ViewSyncCert is complete for view {}!",
-                    self.id, *view_number
-                );
             }
             _ => {}
         };
 
         // We have three cases to consider:
-        let combined = if *view_number > 1 {
-            AndDependency::from_deps(vec![
-                OrDependency::from_deps(vec![AndDependency::from_deps(vec![
-                    payload_commitment_dependency,
-                ])]),
-                OrDependency::from_deps(vec![
-                    // 1. A QCFormed event and QuorumProposalValidated event
-                    AndDependency::from_deps(vec![qc_dependency, proposal_dependency]),
-                    // 2. A timeout cert was received
-                    AndDependency::from_deps(vec![timeout_dependency]),
-                    // 3. A view sync cert was received.
-                    AndDependency::from_deps(vec![view_sync_dependency]),
-                ]),
-            ])
+        let mut secondary_deps = vec![
+            // 2. A timeout cert was received
+            AndDependency::from_deps(vec![timeout_dependency]),
+            // 3. A view sync cert was received.
+            AndDependency::from_deps(vec![view_sync_dependency]),
+        ];
+
+        // 1. A QCFormed event and QuorumProposalValidated event
+        if *view_number > 1 {
+            secondary_deps.push(AndDependency::from_deps(vec![
+                qc_dependency,
+                proposal_dependency,
+            ]));
         } else {
-            AndDependency::from_deps(vec![
-                OrDependency::from_deps(vec![AndDependency::from_deps(vec![
-                    payload_commitment_dependency,
-                ])]),
-                OrDependency::from_deps(vec![
-                    // 1. A QCFormed event and QuorumProposalValidated event
-                    AndDependency::from_deps(vec![qc_dependency]),
-                    // 2. A timeout cert was received
-                    AndDependency::from_deps(vec![timeout_dependency]),
-                    // 3. A view sync cert was received.
-                    AndDependency::from_deps(vec![view_sync_dependency]),
-                ]),
-            ])
-        };
+            secondary_deps.push(AndDependency::from_deps(vec![qc_dependency]));
+        }
+
+        AndDependency::from_deps(vec![
+            OrDependency::from_deps(vec![AndDependency::from_deps(vec![
+                payload_commitment_dependency,
```

Comment:
Nit: Could we simplify this? I don't think we need both `Or` and `And`.

Reply:
Unfortunately we do due to how the type system works. If the heterogenous vector doesn't have the same level of indirection the compiler won't let it through.

Reply:
Also, at the highest level, it requires that an `AndDependency` is the top-level value if any layer uses an And dependency due to how the task is implemented.